### PR TITLE
arch: arm64: Add support for coredump

### PR DESCRIPTION
**Pull request by robertoxmed**
The following pull request adds the following:
* Core files for supporting coredump on ARM64.
* Script to parse the output produced by the tool.

This has been tested using `qemu_cortex_a53` and the tests already included in the tree. Maybe some modification is needed for the tests.

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]
Not sure about the ARM specific parts, but the more general coredump bits look good to me.

**Review by carlocaione — DISMISSED** [review]
good stuff

File: arch/arm64/core/coredump.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+#include <string.h>
+#include <zephyr/debug/coredump.h>
+
+/* Identify the version of this block (in case of architecture changes).
+ * To be interpreted by the target architecture specific block parser.
+ */
+#define ARCH_HDR_VER			1
+
+/* Structure to store the architecture registers passed arch_coredump_info_dump
+ * As callee saved registers are not provided in z_arch_esf_t structure in Bora
```
[the remark this answers is not in the document]
Comment (robertoxmed) [reply arch/arm64/core/coredump.c]: Oops, pushed forced for this small fix. Hope that's fine.

**Review by carlocaione — APPROVED** [review]

**Review by povergoing — APPROVED** [review]

File: arch/arm64/core/coredump.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2022 Huawei Technologies SASU
```
Comment (povergoing) [inline arch/arm64/core/coredump.c:2]: nit: 2022 or 2023?

**Pull request merged**